Guides/JSON-Interop.md

```diff
@@ -0,0 +1,129 @@
+If you are interested in using the `ExtendedJSONEncoder` and `ExtendedJSONDecoder` in your 
+Vapor app instead, you can set them as the default encoder and decoder and thereby allow your 
+application to serialize and deserialize data to/from Extended JSON, rather than the default plain JSON. 
+This is recommended because not all BSON types currently support working with `JSONEncoder` and `JSONDecoder` and 
+also so that you can take advantage of the added type information.
+From the [Vapor Documentation](https://docs.vapor.codes/4.0/content/#override-defaults): 
+you can set the global configuration and change the encoders and decoders Vapor uses by default 
+by doing something like this: 
+
+```swift
+let encoder = ExtendedJSONEncoder()
+let decoder = ExtendedJSONDecoder()
+ContentConfiguration.global.use(encoder: encoder, for: .json)
+ContentConfiguration.global.use(decoder: decoder, for: .json)
+```
+ in your `configure.swift`.
+ 
+ In order for this to work, you will also have to include extensions that ensure conformance to Vapor's 
+ `ContentEncoder` and `ContentDecoder` protocols. The snippets below should be sufficient for doing that.
+ ```swift
+extension ExtendedJSONEncoder: ContentEncoder {
+    public func encode<E>(_ encodable: E, to body: inout ByteBuffer, headers: inout HTTPHeaders) throws 
+        where E: Encodable
+    {
+        headers.contentType = .json
+        try body.writeBytes(self.encode(encodable))
+    }
+}
+ ```
+
+```swift
+extension ExtendedJSONDecoder: ContentDecoder {
+   public func decode<D>(_ decodable: D.Type, from body: ByteBuffer, headers: HTTPHeaders) throws -> D
+       where D: Decodable
+    {
+        let data = body.getData(at: body.readerIndex, length: body.readableBytes) ?? Data()
+        return try self.decode(D.self, from: data)
+    }
+}
+ ```
+
+To see a some example Vapor Apps, check out
+[Examples/VaporExample](https://github.com/mongodb/mongo-swift-driver/tree/master/Examples/VaporExample) or 
+[Examples/ComplexVaporExample](https://github.com/mongodb/mongo-swift-driver/tree/master/Examples/ComplexVaporExample).
+
+## Using `JSONEncoder` and `JSONDecoder` with BSON Types
+
+Currently, some BSON types (e.g. `BSONBinary`) do not support working with encoders and decoders other than those introduced in `swift-bson`, meaning Foundation's `JSONEncoder` and `JSONDecoder` will throw errors when encoding to or decoding from such types. There are plans to add general `Codable` support for all BSON types in the future, though. For now, only `ObjectID` and `BSONDocument` (with only elements that have such support) will work with these and other encoder/decoder pairs. If type information is not required in the output JSON and only types that include a general `Codable` conformance are included in your data, you can use `JSONEncoder` and `JSONDecoder` to produce and ingest JSON data.
```

Comment:
`ObjectID` -> `BSONObjectID`

also, is "`BSONDocument` (with only elements that have such support)" correct? it looks to me like it just follows the same code path as any other BSONValue type and would hit the same errors.

Reply:
Oh you're right--I'm not sure where I got that idea. Fixed.

Reply:
I think we maybe _used_ to do that in the driver, but stopped doing so at some point prior to 1.0.